PWGJE/Tasks/jetfinderQA.cxx

```diff
@@ -435,16 +437,36 @@ struct JetFinderQATask {
       return;
     }
     registry.fill(HIST("h_collisions"), 1.5);
+    fillTrackHistograms(tracks);
+  }
+  PROCESS_SWITCH(JetFinderQATask, processTracks, "QA for charged tracks", false);
+
+  void processTracksWeighted(soa::Join<aod::JCollisions, aod::JMcCollisionLbs>::iterator const& collision,
+                             aod::JMcCollisions const& mcCollisions,
+                             soa::Filtered<JetTracks> const& tracks)
+  {
+    registry.fill(HIST("h_collisions"), 0.5);
+    registry.fill(HIST("h_collisions_weighted"), 0.5, collision.mcCollision().weight());
+    if (!JetDerivedDataUtilities::selectCollision(collision, eventSelection)) {
+      return;
+    }
+    registry.fill(HIST("h_collisions"), 1.5);
+    registry.fill(HIST("h_collisions_weighted"), 1.5, collision.mcCollision().weight());
+    fillTrackHistograms(tracks, collision.mcCollision().weight());
+  }
+  PROCESS_SWITCH(JetFinderQATask, processTracksWeighted, "QA for charged tracks weighted", false);
+
+  void fillTrackHistograms(soa::Filtered<JetTracks> const& tracks, float weight = 1.0)
```

Comment:
can you move this function to line 246 please so its at the end of the other fill functions

Reply:
this has been done for all QA tasks